tests/test_invoice_parser.py

```
@@ -0,0 +1,302 @@
+
+# ── parse_invoice — error cases ───────────────────────────────────────────────
+
+class TestParseInvoiceErrors:
+    def test_missing_file_raises(self, tmp_path):
+        with pytest.raises(InvoiceParseError, match="not found"):
+            parse_invoice(str(tmp_path / "nonexistent.docx"))
+
+    def test_wrong_extension_raises(self, tmp_path):
+        p = tmp_path / "invoice.pdf"
+        p.write_text("data")
+        with pytest.raises(InvoiceParseError, match=".docx"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)